nautobot_ssot/contrib.py

```diff
@@ -0,0 +1,354 @@
+            ) from error
+        return diffsync_model
+
+    @staticmethod
+    def _handle_to_many_relationship(database_object, diffsync_model, parameter_name):
+        """Handle a single one- or many-to-many relationship field.
+
+        one- or many-to-many relationships are type annotated as a list of typed dictionaries. The typed
+        dictionary type expresses, which attributes we are interested in for diffsync.
+        """
+        # Introspect type annotations to deduce which fields are of interest
+        # for this many-to-many relationship.
+        diffsync_field_type = diffsync_model.__annotations__[parameter_name]
+        inner_type = diffsync_field_type.__dict__["__args__"][0]
+        related_objects_list = []
+        # TODO: Allow for filtering, i.e. not taking into account all the objects behind the relationship.
+        for related_object in getattr(database_object, parameter_name).all():
+            dictionary_representation = {
+                field_name: getattr(related_object, field_name) for field_name in inner_type.__annotations__
+            }
+            # Only use those where there is a single field defined, all 'None's will not help us.
+            if any(dictionary_representation.values()):
+                related_objects_list.append(dictionary_representation)
+        return related_objects_list
+
+    @staticmethod
+    def _handle_foreign_key(database_object, parameter_name):
+        """Handle a single foreign key field.
+
+        Given the object from the database as well as the name of parameter in the form of
+        f'{foreign_key_field_name}__{remote_field_name}'
+        return the field at 'remote_field_name' on the object behind the foreign key at 'foreign_key_field_name'.
+
+        Furthermore, 'remote_field_name' may be a series of '__' delimited lookups.
+        """
+        related_model, *lookups = parameter_name.split("__")
+        related_object = getattr(database_object, related_model)
+        # If the foreign key does not point to anything, return None
+        if not related_object:
+            return None
+        for lookup in lookups[:-1]:
+            related_object = getattr(related_object, lookup)
+            # If the foreign key does not point to anything, return None
+            if not related_object:
+                return None
+        # Return the result of the last lookup directly.
+        try:
+            return getattr(related_object, lookups[-1])
+        # If the lookup doesn't point anywhere, check whether it is using the convention for generic foreign keys.
+        except AttributeError:
+            if lookups[-1] in ["app_label", "model"]:
+                return getattr(ContentType.objects.get_for_model(related_object), lookups[-1])
+        return None
+
+
+class NautobotModel(DiffSyncModel):
+    """
+    Base model for any diffsync models interfacing with Nautobot.
```

Comment:
Jsut to make explicit that this is not covering Nautobot via API

```suggestion
    Base model for any diffsync models interfacing with Nautobot via ORM.
```